blockchain/validate.go

```diff
@@ -255,9 +259,14 @@ func CheckTransactionSanity(tx *bchutil.Tx, magneticAnomalyActive bool, scriptFl
 			"%d, max %d", serializedTxSize, MaxTransactionSize)
 		return ruleError(ErrTxTooBig, str)
 	}
-	if magneticAnomalyActive && serializedTxSize < MinTransactionSize {
+
+	if magneticAnomalyActive && serializedTxSize < MagneticAnomalyMinTransactionSize {
+		minTxSize := MagneticAnomalyMinTransactionSize
+		if upgrade9Active {
```

Comment:
I think in rational cases this chained logic works. However, to be safe, I would personally put it outside as its own test. For example, if for some crazy reason someone runs the engine with `magneticAnomalyActive` flag disabled, then the chained logic would not detect a tx too small for `upgrade9`.

:shrug: It doesn't matter in practical terms, right?

Reply:
I get that it makes the ordering explicit. If that's the intent, then maybe a note about ordering would be nice.